vcenter/vcenter.py

```diff
@@ -449,6 +441,88 @@ def stop(self, uuid):
                 settings.raven.captureException(exc_info=True)
                 self.__sleep_between_tries()
 
+    def take_screenshot(self, uuid):
+        """
+        Takes screenshot of VM and saved it in datastore
+        :param uuid: machine uuid
+        :return: tuple; name of datastore (where screenshot is saved) and path to screenshot in datastore
+        """
+        self.__logger.debug('-> take_screenshot()')
+        self.__check_connection()
+        vm = self.content.searchIndex.FindByUuid(None, uuid, True)
+        if vm is None:
+            raise Exception(f'machine {uuid} not found')
+
+        self.__logger.debug(f'found vm: {vm.config.uuid}')
+        screenshot_task = vm.CreateScreenshot_Task()
+        self.wait_for_task(screenshot_task)
+        result_path = screenshot_task.info.result
+        if result_path is None:
+            raise RuntimeError('Taking screenshot failed!')
+
+        # can't we just use self.destination_datastore.info.name ?
+        datastore_name, screenshot_path = result_path.split(' ')
+        datastore_name = datastore_name.lstrip('[').rstrip(']')
+        self.__logger.debug(f'<- take_screenshot: {datastore_name}, {screenshot_path}')
+        return datastore_name, screenshot_path
+
+    # TODO rewrite others to use this one
+    def __get_objects_list_from_container(self, container, object_type):
+
+        result = []
+        object_view = None
+        try:
+            object_view = self.content.viewManager.CreateContainerView(
+                    container,
+                    [object_type],
+                    True)
+            result = list(object_view.view)
+        except vmodl.fault.ManagedObjectNotFound:
+            self.__logger.warn('vmodl.fault.ManagedObjectNotFound has occured')
+        except Exception:
+            settings.raven.captureException(exc_info=True)
+        finally:
+            if object_view is not None:
+                object_view.Destroy()
+
+        return result
```

Comment:
this is a generic function for getting object from view, should return either list of objects as requested, or empty list (in case of failure)